x/marker/types/send_restrictions.go

```diff
@@ -36,27 +36,28 @@
 	return isBool && bypass
 }
 
-// WithTransferAgent returns a new context that contains the provided marker transfer agent.
-func WithTransferAgent[C context.Context](ctx C, transferAgent sdk.AccAddress) C {
+// WithTransferAgents returns a new context that contains the provided marker transfer agent.
+// This will overwrite any existing transfer agents in the context.
+func WithTransferAgents[C context.Context](ctx C, transferAgents ...sdk.AccAddress) C {
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
-	sdkCtx = sdkCtx.WithValue(transferAgentKey, transferAgent)
+	sdkCtx = sdkCtx.WithValue(transferAgentKey, transferAgents)
 	return context.Context(sdkCtx).(C)
 }
 
-// WithoutTransferAgent returns a new context with a nil marker transfer agent.
-func WithoutTransferAgent[C context.Context](ctx C) C {
+// WithoutTransferAgents returns a new context without any marker transfer agents.
+func WithoutTransferAgents[C context.Context](ctx C) C {
```

Comment:
It doesn't really matter because `WithTransferAgents` writes the value without trying to read it. And in `GetTransferAgents`, it returns `nil` when the context either doesn't have a `transferAgentKey`, or has one, but it's the wrong type, or has a nil value the right type.

But it might later cause confusion, so I'll fix this.